dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java

```diff
@@ -42,10 +46,58 @@
 public final class GitHubHelper {
 
     private static final Logger LOG = LoggerFactory.getLogger(GitHubHelper.class);
+    public static final String BRANCHNAME_FOR_BOT = "feature/add_dockstore_yml";
+
+    public static final String DOCKSTORE_BOT_PR_TEXT = """
+           The dockstore-bot has guessed at a .dockstore.yml for your repository.
+           
+           Please review, make appropriate changes, add any [additional keys](https://docs.dockstore.org/en/stable/assets/templates/template.html) and merge in order to complete integration with Dockstore, allowing Dockstore to keep informed of new changes to your workflow(s), notebook(s), or tool(s).
+           """;
+
+    public static final String DOCKSTORE_BOT_PR_TITLE =  "dockstore-bot inferred .dockstore.yml";
 
     private GitHubHelper() {
     }
 
+    public static URL createForkPlusPR(String inferredDockstoreYml, GitHub gitHub, String repositoryName, String targetBranch) {
+        return createForkPlusPR(inferredDockstoreYml, gitHub, repositoryName, targetBranch, BRANCHNAME_FOR_BOT, DOCKSTORE_BOT_PR_TITLE, DOCKSTORE_BOT_PR_TEXT);
+    }
+
+    /**
+     * Experimental building block, given a repository, create a fork of it (to the personal organization for the robot user) and a PR to it
+     * @param inferredDockstoreYml should be the only real change in the PR
+     * @param gitHub    The GitHub API
+     * @param repositoryName    Name of the GitHub repository (e.g. dockstore/lambda)
+     * @param targetBranch    Name of the branch in the GitHub repository to make a PR against (e.g. develop)
+     * @param branchForBot  Name of the branch in the fork
```

Comment:
Do we envision that we'll infer .dockstore.yml for multiple branches of a repository? If so, we could suffix the target branch name to the name of the branch in the fork like `feature/add_dockstore_yml_main`

Reply:
Thinking no, I'd probably prefer just the one PR to the default branch if possible 
(seems simpler, avoid merge conflicts, etc.)